### PR TITLE
Folder-Structure

### PR DESCRIPTION
**Pull request by Danneytrieu**
I made a few changes and tried to establish a decent way to organize everything. Certain practices won't make much sense now but as the projects grow, they will come in handy as we try to navigate and change contents/styles/themes. Here is my approach:

- theme.js -> each theme will hold 2 obj, `color` can be used to apply to text, while `background` applies to container, card or button. Also used theme provider to access these objects globally. Yes, some colors do repeat but the color collection might change in the future and it will be easier to adopt more themes later. 
- global-style.tsx ->  added a few reusable template component 
- SignUp.elements.tsx -> hold all related styled components and styled icon that belong to SignUp 
- SignUp.tsx -> import everything from SignUp.elements.tsx and applied separately to prevent future error prone. 

Please let me know if something doesn't make sense or better way to structure and better practices.

### CODE REVIEW

**Review by theodorusandi — CHANGES_REQUESTED** [review]
I think the two global components (Container and Card) should not be inside global style since its only for styling and we don't want to import all from global. Would be better to put it just under components folder as two independent files.

**Review by jpatrici — COMMENTED** [review]
I think for the common components that can be used widely, we should still put it under `components` folder. So things like `Container`, `Section` can be put under `common` folder. Same thing with the `Auth` as from my comment, they can be put under `components`.

Maybe we can follow this https://github.com/withspectrum/spectrum/tree/alpha/src, so all the components are inside `components` folder. Pages are inside `views` folder, and they import the common components from `components` folder. Any specific styling just for the page, can go inside that page folder. For example
```
/components
   Container.tsx
   Section.tsx
   Auth.tsx
/pages
   /SignUp
      SignUp.tsx
      SignUp.styled.tsx
   /SignIn
      SignIn.tsx
```
I also agree with @theodorusandi that they shouldn't be in the global style file, and to avoid doing `import *`

File: src/components/SignUp/SignUp.elements.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  margin: 0 20px;
+`;
+export const EmailIcon = styled(HiMail)`
+  font-size: 1rem;
+  margin: 0 20px;
+`;
+export const PasswordIcon = styled(RiLockPasswordFill)`
+  font-size: 1rem;
+  margin: 0 20px;
+`;
```
[the remark this answers is not in the document]
Comment (Danneytrieu) [reply src/components/SignUp/SignUp.elements.tsx:41]: interesting! I was looking for something like this

File: src/components/SignUp/SignUp.elements.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import styled from 'styled-components';
+import { FaUserAlt, FaTwitterSquare } from 'react-icons/fa';
+import { HiMail } from 'react-icons/hi';
+import { RiLockPasswordFill } from 'react-icons/ri';
+import { Link } from 'react-router-dom';
+import * as global from '../../global-style';
```
[the remark this answers is not in the document]
Comment (Danneytrieu) [reply src/components/SignUp/SignUp.elements.tsx:6]: gotcha

**Review by RDiMag — APPROVED** [review]

**Review by theodorusandi — APPROVED** [review]
Looks good to me

**Review by jpatrici — APPROVED** [review]
thanks Danney!

**Pull request merged**